solidity/contracts/transformers/ProtocolTokenWrapperTransformer.sol

```diff
@@ -46,46 +47,68 @@ contract ProtocolTokenWrapperTransformer is BaseTransformer {
   }
 
   /// @inheritdoc ITransformer
-  // slither-disable-next-line arbitrary-send
   function transformToUnderlying(
     address _dependent,
     uint256 _amountDependent,
     address _recipient
   ) external returns (UnderlyingAmount[] memory) {
-    IERC20(_dependent).safeTransferFrom(msg.sender, address(this), _amountDependent);
-    IWETH9(_dependent).withdraw(_amountDependent);
-    payable(_recipient).transfer(_amountDependent);
+    _takeAndUnwrap(IWETH9(_dependent), _amountDependent, _recipient);
     return _toUnderylingAmount(PROTOCOL_TOKEN, _amountDependent);
   }
 
   /// @inheritdoc ITransformer
-  // slither-disable-next-line arbitrary-send
   function transformToDependent(
     address _dependent,
     UnderlyingAmount[] calldata _underlying,
     address _recipient
   ) external payable returns (uint256 _amountDependent) {
     _amountDependent = _underlying[0].amount;
-    IWETH9(_dependent).deposit{value: _amountDependent}();
-    IERC20(_dependent).safeTransfer(_recipient, _amountDependent);
+    _wrapAndTransfer(IWETH9(_dependent), _amountDependent, _recipient);
   }
 
   /// @inheritdoc ITransformer
   function transformToExpectedUnderlying(
     address _dependent,
     UnderlyingAmount[] calldata _expectedUnderlying,
     address _recipient
-  ) external returns (uint256 _spentDependent) {}
+  ) external returns (uint256 _spentDependent) {
+    _spentDependent = _expectedUnderlying[0].amount;
+    _takeAndUnwrap(IWETH9(_dependent), _spentDependent, _recipient);
+  }
 
   /// @inheritdoc ITransformer
   function transformToExpectedDependent(
     address _dependent,
     uint256 _expectedDependent,
     address _recipient
-  ) external payable returns (UnderlyingAmount[] memory _spentUnderlying) {}
+  ) external payable returns (UnderlyingAmount[] memory _spentUnderlying) {
+    _wrapAndTransfer(IWETH9(_dependent), _expectedDependent, _recipient);
+    return _toUnderylingAmount(PROTOCOL_TOKEN, _expectedDependent);
+  }
```

Comment:
I don't remember if we talked about this... but... What if... `msg.value != _expectedDependent`? Should we change the behavior? Should we... Just take it as dust? Shouldn't happend but....

Reply:
We are not looking at `msg.value`, because we want to support multicalls. If someone sends more ether than they specify in the parameters, then we will be able to take it out by using `CollectableDust`